make/env.cmake

```diff
@@ -159,11 +161,15 @@ function(ADD_CONDA_PACKAGE)
         ${binary_upper} ${${binary_upper}}
         ${binary_upper}_TARGET ${TARGET})
     endforeach()
-  elseif()
-    set(${binary_upper} ${NAME})
-    replace_with_env_if_set(${binary_upper})
-    set_target_properties(env PROPERTIES ${binary_upper} ${${binary_upper}})
-    set_target_properties(env PROPERTIES ${binary_upper}_TARGET "")
+  else()
+    foreach(OUTPUT ${ADD_CONDA_PACKAGE_PROVIDES})
+      string(TOUPPER ${OUTPUT} binary_upper)
+      set(${binary_upper} ${OUTPUT})
+      replace_with_env_if_set(${binary_upper})
+      set_target_properties(env PROPERTIES
```

Comment:
If you want this to support finding programs on the path, than if replace_with_env_if_set fails to find the executable, should use find_program?

Reply:
So if replace_with_env_if_set does not clear the variable if it found no environment to replace it with. Which resulted in the variable just being the binary name to execute.

I am not very familiar with CMake, but I could not find a function that would provide the full path to a program that worked like find_program. Do you know of a way to get the full path `find_program()` found the program at? As doing such would make it easy to solve the use of these property values as dependencies.

Reply:
I had been testing with find_program after the set, which in turn was skipping the searching and just leaving the variable to the value of the set.